src/socket/tcp.rs

```diff
@@ -3656,33 +3656,25 @@ mod test {
     #[test]
     fn test_buffer_wraparound_tx() {
         let mut s = socket_established();
-        s.tx_buffer = SocketBuffer::new(vec![0; 6]);
-        assert_eq!(s.send_slice(b"abc"), Ok(3));
+        s.tx_buffer = SocketBuffer::new(vec![b'.'; 9]);
+        assert_eq!(s.send_slice(b"xxxyyy"), Ok(6));
+        assert_eq!(s.tx_buffer.dequeue_many(3), &b"xxx"[..]);
```

Comment:
This is not any better...

Reply:
Why touch the internal `tx_buffer` you ask? The `read_at` pointer must be initialized so that the check still makes sense when the optimization is merged.